File: web/help.go
// Townsourced
// Copyright 2015 Tim Shannon. All rights reserved.

package web

import (
	"net/http"

	log "git.townsourced.com/townsourced/logrus"
	"git.townsourced.com/townsourced/townsourced/app"
	"git.townsourced.com/townsourced/townsourced/data"
)

func helpTemplate(w http.ResponseWriter, r *http.Request, c context) {
	key := data.NewKey(c.params.ByName("key"))

	help, err := app.HelpGet(key)
	if errHandledPage(err, w, r, c) {
		return
	}

	err = w.(*templateWriter).execute("HELP", help)

	if err != nil {
		log.Errorf("Error executing help template: %s", err)
	}
}

File: web/src/js/help.js
// Townsourced
// Copyright 2015 Tim Shannon. All rights reserved.
/* jshint  esnext: true, strict: true */

import Ractive from "./ts/ractiveInit";
import Page from "./components/page";

import {
    htmlPayload,
}
from "./ts/util";

$(document).ready(function() {
    "use strict";

    var r = new Ractive({
        el: "body",
        template: "#tMain",
        components: {
            page: Page,
        },
		data: function() {
			return {
				help: htmlPayload(),
				processed: null,	
			};
		},
		oncomplete: function() {
			this.set("processed", ts.processMessage(this.get("help.document")));
		},
    });

});
